ouroboros-network/src/Ouroboros/Network/Diffusion/P2P.hs

```diff
@@ -0,0 +1,1260 @@
+-- 'InitiatorResponderMode'.
+--
+data ConnectionManagerDataInMode (mode :: MuxMode) where
+    CMDInInitiatorMode
+      :: ConnectionManagerDataInMode InitiatorMode
+
+    CMDInInitiatorResponderMode
+      :: Server.ControlChannel IO
+          (Server.NewConnection
+            SockAddr
+            (Handle InitiatorResponderMode SockAddr ByteString IO () ()))
+      -> StrictTVar IO Server.InboundGovernorObservableState
+      -> ConnectionManagerDataInMode InitiatorResponderMode
+
+
+--
+-- Node-To-Client type aliases
+--
+-- Node-To-Client diffusion is only used in 'ResponderMode'.
+--
+
+type NodeToClientHandle =
+    Handle ResponderMode LocalAddress ByteString IO Void ()
+
+type NodeToClientHandleError =
+    HandleError ResponderMode NodeToClientVersion
+
+type NodeToClientConnectionHandler =
+    ConnectionHandler
+      ResponderMode
+      (ConnectionHandlerTrace NodeToClientVersion NodeToClientVersionData)
+      LocalSocket
+      LocalAddress
+      NodeToClientHandle
+      NodeToClientHandleError
+      (NodeToClientVersion, NodeToClientVersionData)
+      IO
+
+type NodeToClientConnectionManagerArguments =
+    ConnectionManagerArguments
+      (ConnectionHandlerTrace NodeToClientVersion NodeToClientVersionData)
+      LocalSocket
+      LocalAddress
+      NodeToClientHandle
+      NodeToClientHandleError
+      (NodeToClientVersion, NodeToClientVersionData)
+      IO
+
+type NodeToClientConnectionManager =
+    ConnectionManager
+      ResponderMode
+      LocalSocket
+      LocalAddress
+      NodeToClientHandle
+      NodeToClientHandleError
+      IO
+
+--
+-- Node-To-Node type aliases
+--
+-- Node-To-Node diffusion runs in either 'InitiatorMode' or 'InitiatorResponderMode'.
+--
+
+type NodeToNodeHandle (mode :: MuxMode) a =
+    Handle mode SockAddr ByteString IO () a
+
+type NodeToNodeHandleError (mode :: MuxMode) =
+    HandleError mode NodeToNodeVersion
+
+type NodeToNodeConnectionHandler (mode :: MuxMode) a =
+    ConnectionHandler
+      mode
+     (ConnectionHandlerTrace NodeToNodeVersion NodeToNodeVersionData)
+     Socket
+     SockAddr
+      (NodeToNodeHandle mode a)
+      (NodeToNodeHandleError mode)
+     (NodeToNodeVersion, NodeToNodeVersionData)
+     IO
+
+type NodeToNodeConnectionManagerArguments (mode :: MuxMode) a =
+    ConnectionManagerArguments
+      (ConnectionHandlerTrace NodeToNodeVersion NodeToNodeVersionData)
+      Socket
+      SockAddr
+      (NodeToNodeHandle mode a)
+      (NodeToNodeHandleError mode)
+      (NodeToNodeVersion, NodeToNodeVersionData)
+      IO
+
+type NodeToNodeConnectionManager (mode :: MuxMode) a =
+    ConnectionManager
+      mode
+      Socket
+      SockAddr
+      (NodeToNodeHandle mode a)
+      (NodeToNodeHandleError mode)
+      IO
+
+--
+-- Governor type aliases
+--
+
+type NodeToNodePeerConnectionHandle (mode :: MuxMode) a =
+    PeerConnectionHandle
+      mode
+      SockAddr
+      ByteString
+      IO () a
```

Comment:
This belongs to the previous section, but it's a mistake in the `p2p-master`, so let's leave it for now.